File: Modules/Common/Techs/SIE.Web.Tech/Processs/Commands/SelectProcessUserGroupCommand.cs
using SIE.Domain;
using SIE.Tech.Processs;
using SIE.Web.Command;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SIE.Web.Tech.Processs.Commands
{
    public class SelectProcessUserGroupCommand: ViewCommand
    {
        /// <summary>
        /// 选择工序命令执行方法
        /// </summary>
        /// <param name="args">视图参数</param>
        /// <param name="scope">作用域</param>
        /// <returns>执行结果</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            var processUserGroupList = args.Data.ToJsonObject<List<ProcessUserGroup>>();
            Check.NotNullOrEmpty(processUserGroupList, nameof(processUserGroupList));
            if (processUserGroupList == null || processUserGroupList.Count == 0)
            {
                throw new ArgumentNullException("{0}数据参数不能为空".L10nFormat(nameof(processUserGroupList)));
            }
            RT.Service.Resolve<ProcessController>().SelectSaveProcessUserGroup(processUserGroupList);
            //foreach (var item in processUserGroupList)
            //{
            //    var processUserGroup = new ProcessUserGroup();
            //    processUserGroup.UserGroupId = item.UserGroupId;
            //    processUserGroup.ProcessId = item.ProcessId;
            //    RF.Save(processUserGroup);
            //}

            return true;
        }
    }
}

File: Modules/Common/Techs/SIE.Web.Tech/Processs/Commands/SelectProcessUserGroupCommand.js
/**
  * 选择工序命令
  * 
*/
SIE.defineCommand('SIE.Web.Tech.Processs.Commands.SelectProcessUserGroupCommand', {
    extend: 'SIE.cmd.LookupCommandBase',
    userConfig: {
        dataParams: { specKeyPrototyName: 'ProcessId', targetClassName: 'SIE.Tech.Processs.Process' },
    },
    meta: { text: "选择", group: "edit", iconCls: "icon-PlaylistCheck icon-blue" },
    /**
     * 重写保存方法
     * @param {win} 窗口
     */
    save: function (win) {
        var me = this;
        var indata = {};
        var selections = me._targetSelectItems.items;
        if (selections && selections.length > 0) {
            var operationDatas = [];
            SIE.each(selections, function (item) {
                var processId = item.getId();
                if (me._sourceViewSelectItems.indexOf(processId) === -1) {
                    var processUserGroup = { UserGroupId: me._sourceId, ProcessId: processId };
                    operationDatas.push(processUserGroup);
                }
            });
            indata = operationDatas;
            me._targetView.execute({
                data: indata,
                success: function (res) {
                    win.close();
                    me._ownerView.loadChildData(true);
                }
            }, me._ownerView);
        }
        else {
            SIE.Msg.showWarning('没有可提交的数据'.t());
        }
    }
});
